### ISSUE TITLE
Conky 1.11.1 crashing with SIGILL on FreeBSD head

### ISSUE THREAD

**Issue by madpilot78** [brndnmtthws/conky#696]
I'm testing the latest conky release to update the FreeBSD port, but I'm getting a strange crash I can't really fix. It crashes just after startup.

I built it with debugging symbols enabled and disabling all optmizations and I get this information out of gdb:

```
Core was generated by `conky'.
Program terminated with signal SIGILL, Illegal instruction.
#0  0x00000000002c9c56 in update_total_processes ()
    at /wrkdirs/usr/ports/sysutils/conky/work/conky-1.11.0/src/freebsd.cc:256
256	/wrkdirs/usr/ports/sysutils/conky/work/conky-1.11.0/src/freebsd.cc: No such file or directory.
[Current thread is 1 (LWP 101924)]
(gdb) bt
#0  0x00000000002c9c56 in update_total_processes ()
    at /wrkdirs/usr/ports/sysutils/conky/work/conky-1.11.0/src/freebsd.cc:256
#1  0x0000000000269fa6 in legacy_cb::work (this=0x801468ce0)
    at /wrkdirs/usr/ports/sysutils/conky/work/conky-1.11.0/src/text_object.h:82
#2  0x00000000002c54fc in conky::priv::callback_base::start_routine (
    this=0x801468ce0)
    at /wrkdirs/usr/ports/sysutils/conky/work/conky-1.11.0/src/update-cb.cc:128
#3  0x00000000002c870d in std::__1::__invoke<void (conky::priv::callback_base::*)(), conky::priv::callback_base*, , void> (
    __f=@0x8015ba6a8: (void (conky::priv::callback_base::*)(conky::priv::callback_base * const)) 0x2c5490 <conky::priv::callback_base::start_routine()>, 
    __a0=@0x8015ba6b8: 0x801468ce0) at /usr/include/c++/v1/type_traits:4423
#4  std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (conky::priv::callback_base::*)(), conky::priv::callback_base*, 2ul> (__t=...)
    at /usr/include/c++/v1/thread:342
#5  std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (conky::priv::callback_base::*)(), conky::priv::callback_base*> > (__vp=0x8015ba6a0)
    at /usr/include/c++/v1/thread:352
#6  0x000000080086a775 in ?? () from /lib/libthr.so.3
#7  0x0000000000000000 in ?? ()
Backtrace stopped: Cannot access memory at address 0x7fffdf9fb000
```

Testing a binary built int he same way on a FreeBSD 11.2 VM works fine, but I'd like to make sure it works in all cases before updating the official port.

Any idea or indications where to look at?

Thanks!

**Comment by madpilot78** [comment]
Just tested the branch you pointed me to, but I get the same exact failure.

I don't know where to look at to get some more insight, I'll think about it in the next days.

**Comment by lasers** [comment]
Maybe you can git bisect this and find the first bad commit. That might lead us somewhere.

**Comment by madpilot78** [comment]
@lasers Thanks, yes, I am planning to do that, but I've been a little busy recently and could not get to it.

I'll try to do that soon and report back.

**Comment by madpilot78** [comment]
After performing the bisect I discovered the faulty commit is e91b91d.

Looking at it I noticed this strange change removing a return 0 statement from a function returning an int.

Adding back the return statement avoids the crash.

**Comment by brndnmtthws** [comment]
Yikes, sounds like I messed up. I'll get a PR out soon.

**Comment by brndnmtthws** [comment]
I do recall what happened here: I started refactoring that code to make another change, then I abandoned that work, but didn't properly revert my staging area.

**Comment by lasers** [comment]
Closed via #750. You rock! :guitar:
